AFC.py

```diff
@@ -650,13 +653,17 @@ def cmd_CHANGE_TOOL(self, gcmd):
             self.toolhead.wait_moves()
             if self.is_printing() and not self.is_paused():
                 self.change_tool_pos = None
+                # Restore state
+                self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
 
     cmd_RESTORE_CHANGE_TOOL_POS_help = "change filaments in tool head"
     def cmd_RESTORE_CHANGE_TOOL_POS(self, gcmd):
         if self.change_tool_pos:
             restore_pos = self.change_tool_pos[:3]
             self.toolhead.manual_move(restore_pos, self.tool_unload_speed)
             self.toolhead.wait_moves()
+            # Restore previous state
+            self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
```

Comment:
One thing I forgot in my change is we should clear the saved position variable after restoring to avoid issues around it still existing if this macro gets called later out of context of a too. change. Any chance you can toss that in here?

```suggestion
            self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
            self.change_tool_pos = None
```